app/src/main/java/de/dhbw/corona_world_app/ui/map/MapFragment.java

```diff
@@ -169,9 +174,37 @@ public void onSlide(@NonNull View bottomSheet, float slideOffset) {
         WebView myWebView = root.findViewById(R.id.map_web_view);
         WebSettings webSettings = myWebView.getSettings();
 
+        button.setOnClickListener(v -> {
+            if(mapViewModel.getCurrentResolution().equals(MapData.Resolution.WOLRD)){
+                root.findViewById(R.id.bottomSheetButton).setVisibility(View.GONE);
+                mapViewModel.switchResolution(MapData.Resolution.GERMANY);
+                try {
+                    mapViewModel.initGermany();
+                } catch (IOException | ClassNotFoundException | InterruptedException | ExecutionException | JSONException e) {
+                    handleException(e);
+                }
+            } else {
+                root.findViewById(R.id.bottomSheetButton).setVisibility(View.INVISIBLE);
+                mapViewModel.switchResolution(MapData.Resolution.WOLRD);
+                try {
+                    mapViewModel.initCountryList();
+                } catch (InterruptedException | ExecutionException | IOException | JSONException | ClassNotFoundException e) {
+                    handleException(e);
+                }
+            }
+        });
+
```

Comment:
consider using a abstract Fragment and implementing it for both germany and world. See StatisticCallRecyclerViewFragment and its implementations History Fragment and Favourite Fragment

Reply:
It was planned to have Germany Map and World Map in two different Fragments and not in one with a switch button. You dont need to worry about the Design/Navigation, just create the Germany Map as a seperate Fragment and i handle the integration into the current Design

Reply:
I think an additional fragment for each new country is not very smart. If we were to add other countries in the future that would take a tremendously higher effort to integrate into the app. 
The generelization you were on about is no bad idea, the Viewmodel's methods could very well be brought together so that just one call ramains in the fragment.

Reply:
If you made an abstract Fragment, integrating a new country would not be much effort as you would just need to implement the abstract fragment and add it to navigation.

Reply:
Ehm, yes and that is a bit more than I have to do with my approach, where you forget about the fragment and just have to implement the datastructure and api (this you would have to do too in your approach btw)

Reply:
Lets move this to private and post the solution later here

Reply:
After disscussion, the germany map will be put in a seperate fragment due to navigation issues

Reply:
Is now in seperate fragment